state/state.go

```diff
@@ -34,19 +42,128 @@ type DownloadedObject struct {
 	Object, Filename string
 }
 
+type DynamoDBStater struct {
+	Session          *session.Session
+	Service          string
+	BackfillInterval time.Duration
+}
+
+func NewDynamoDBStater(session *session.Session, service string, backfill int) *DynamoDBStater {
+
+	return &DynamoDBStater{
+		Session:          session,
+		Service:          service,
+		BackfillInterval: time.Hour * time.Duration(backfill),
+	}
+}
+
+// Used for unmarshaling and adding objects to DynamoDB
+type Record struct {
+	S3Object string
+	Time     time.Time
+	TTL      int64
+}
+
+// list of processed objects
+func (d *DynamoDBStater) ProcessedObjects() (map[string]time.Time, error) {
+	objs := make(map[string]time.Time)
+
+	var records []Record
+
+	svc := dynamodb.New(d.Session)
+	err := svc.ScanPages(&dynamodb.ScanInput{
+		TableName: aws.String(DynamoTableName),
+	}, func(logs *dynamodb.ScanOutput, last bool) bool {
+		recs := []Record{}
+
+		err := dynamodbattribute.UnmarshalListOfMaps(logs.Items, &recs)
+
+		// break out of function
+		if err != nil {
+			logrus.WithField("error", err).Debug("Failed to unmarshal DynamoDB Scan Items")
+			return false
+		}
+
+		records = append(records, recs...)
+		for _, rec := range records {
+			// break out of the scan, we've reached the end of our
+			// backfill interval
+			if time.Since(rec.Time) > d.BackfillInterval {
+				return false
+			}
+		}
+
+		return true
+	})
+
+	if err != nil {
+		return objs, fmt.Errorf("Error scanning DynamoDB, %v", err)
+	}
+
+	for _, record := range records {
+		objs[record.S3Object] = record.Time
+	}
+
+	return objs, nil
+}
+
+func (d *DynamoDBStater) SetProcessed(s3object string) error {
+
+	// add access log information
```

Comment:
I'm not sure what this comment means. Creating the new dynamodb.Session isn't adding access log info - is that the whole function? (then maybe move to the line before the `func` definition)  Maybe just delete this line?